traffic_portal/app/src/common/modules/form/topology/edit/FormEditTopologyController.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var FormEditTopologyController = function(topologies, cacheGroups, $scope, $controller, $uibModal, $anchorScroll, locationUtils, topologyService, messageModel, topologyUtils) {
+
+	// extends the FormTopologyController to inherit common methods
+	angular.extend(this, $controller('FormTopologyController', { topology: topologies[0], cacheGroups: cacheGroups, $scope: $scope }));
+
+	let deleteTopology = function(topology) {
+		topologyService.deleteTopology(topology)
+			.then(function() {
+				messageModel.setMessages([ { level: 'success', text: 'Topology deleted' } ], true);
+				locationUtils.navigateToPath('/topologies');
+			});
+	};
+
+	$scope.topologyName = angular.copy($scope.topology.name);
+
+	$scope.settings = {
+		isNew: false,
+		saveLabel: 'Update'
+	};
+
+	$scope.save = function(name, description, topologyTree) {
+		let normalizedTopology = topologyUtils.getNormalizedTopology(name, description, topologyTree);
+		topologyService.updateTopology(normalizedTopology).
+			then(function() {
+				messageModel.setMessages([ { level: 'success', text: 'Topology updated' } ], false);
+			});
+	};
+
+	$scope.confirmDelete = function(topology) {
+		let params = {
+			title: 'Delete Topology: ' + topology.name,
+			key: topology.name
+		};
+		let modalInstance = $uibModal.open({
+			templateUrl: 'common/modules/dialog/delete/dialog.delete.tpl.html',
+			controller: 'DialogDeleteController',
+			size: 'md',
+			resolve: {
+				params: function () {
+					return params;
+				}
+			}
+		});
+		modalInstance.result.then(function() {
+			deleteTopology(topology);
+		}, function () {
+			// do nothing
+		});
```

Comment:
doing nothing could swallow an exception, can we `console.error` or `console.warn` instead?